esp32s3: can't build smp_srv sample due to mcuboot assert issue

**Describe the bug**
Cannot build and run `samples/subsys/mgmt/mcumgr/smp_svr` as build fails.

**To Reproduce**
Steps to reproduce the behavior:
1. west build -p -b esp32s3_devkitc/esp32s3/procpu samples/subsys/mgmt/mcumgr/smp_svr --sysbuild

**Expected behavior**
Build fully working.

**Impact**
Showstopper as smp_svr can't be executed.

**Logs and console output**
```
: in function `boot_set_next':
/zephyr/bootloader/mcuboot/boot/bootutil/src/bootutil_public.c:551: undefined reference to `mcuboot_assert_handler'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: cmake --build 
```


**Environment (please complete the following information):**

 - Linux/Ubuntu
 - SDK 0.17.0
 - v4.0.0-rc3

--- a/west.yml
+++ b/west.yml
@@ -157,7 +157,7 @@ manifest:
       groups:
         - hal
     - name: hal_espressif
-      revision: 174547ef6a97dafcd6786ecd171cc701f5c0893b
+      revision: 07ff57e8d197765652b7819b297415d859ed7815
       path: modules/hal/espressif
       west-commands: west/west-commands.yml
       groups:

